fetchcode/api.py

```diff
@@ -14,9 +14,10 @@
 # CONDITIONS OF ANY KIND, either express or implied. See the License for the
 # specific language governing permissions and limitations under the License.
 
+import shutil
 import tempfile 
-
-import requests
```

Comment:
Are you really dropping using requests?

Reply:
Yes, I thought I should use urllib since it can support both http and ftp urls

Reply:
hum... I am not sure that the convenience is worth the possible trouble. "requests" is a tad more robust than the plain urllib AFAIK. Also have you considered https://pypi.org/project/ftputil/ also for FTP?

Reply:
what about we use ftplib ?